src/chttpd/src/chttpd.erl

```diff
@@ -236,7 +236,15 @@ handle_request_int(MochiReq) ->
     % for the path, use the raw path with the query string and fragment
     % removed, but URL quoting left intact
     RawUri = MochiReq:get(raw_path),
-    {"/" ++ Path, _, _} = mochiweb_util:urlsplit_path(RawUri),
+    {Path0, _, _} = mochiweb_util:urlsplit_path(RawUri),
+    Path =
+        case Path0 of
+            "/" ++ ValidPath ->
+                ValidPath;
+            InvalidPath ->
+                couch_log:info("~p shutdown from invalid path ~p", [?MODULE, InvalidPath]),
```

Comment:
I am slightly worried about dumping any random bytes they put in the path into the logs.

Maybe we can just do `couch_stats:increment_counter([couchdb, httpd, aborted_requests])`?

Reply:
Oh yes, that's a much better idea! Pushed a fixup for that.